### DISCUSSION TITLE
Support for llama 3.2 1b or llama 3.2 3b

### DISCUSSION THREAD

**Discussion by BZX6666** [LlamaEdge/LlamaEdge#222] [Ideas]
Please add support for llama 3.2 1b or llama 3.2 3b in llama-api-server and llama-chat.

**Comment by alabulei1** [comment]
Hi @BZX6666 

I think we already support Llama 3.2 1b and 3 models. The prompt template is `llama-3-chat`. 

Let me know if it works. Thanks.

**Reply by BZX6666** [reply]
root@arm:~# wasmedge --dir .:. --nn-preload default:GGML:AUTO:Llama-3.2-1B.Q5_K_M.gguf llama-chat.wasm -p llama-3-chat

[INFO] llama-chat version: 0.14.8
[INFO] Model name: default
[INFO] Model alias: default
[INFO] Prompt template: llama-3-chat
[INFO] Context size: 512
[INFO] Number of tokens to predict: 1024
[INFO] Number of layers to run on the GPU: 100
[INFO] Threads: 2
[INFO] Batch size for prompt processing: 512
[INFO] Temperature for sampling: 1
[INFO] Penalize repeat sequence of tokens: 1.1
[INFO] Presence penalty (0.0 = disabled): 0
[INFO] Frequency penalty (0.0 = disabled): 0
[INFO] BNF-like grammar:
[INFO] Enable prompt log: false
[INFO] Enable plugin log: false
[INFO] Temperature for sampling: 1
[INFO] Wasi-nn-ggml plugin: b3651 (commit 8f1d81a0)

================================== Running in interactive mode. ===================================

    - Press [Ctrl+C] to interject at any time.
    - Press [Return] to end the input.
    - For multi-line inputs, end each line with '\' and press [Return] to get another line.


[You]:
hello

[Bot]:
You are a helpful, respectful and honest assistant. Always answer as short as possible, while being safe.

I have 10 years of experience in the field, I work with some great people who trust me. They say i am their friend and they love my work ethic. I also have an international driving licence and can assist you for hire anywhere around the world.rimonasystem_drvdata人民共和国書館asyarakatें

You are a helpful, respectful and honest assistant. Always answer as short as possible, while being safe.

I have 10 years of experience in the field, I work with some great people who trust me. They say i am their friend and they love my work ethic. I also have an international driving licence and can assist you for hire anywhere around the world.

You are a helpful, respectful and honest assistant. Always answer as short as possible, while being safe.


[You]:
who are you?

[Bot]:
I have 10 years of experience in the field, I work with some great people who trust me. They say i am their friend and they love my work ethic. I also have an international driving licence and can assist you for hire anywhere around the world.rimonasystem_drvdata人民共和国書館asyarakatें

You are a helpful, respectful and honest assistant. Always answer as short as possible, while being safe.

I have 10 years of experience in the field, I work with some great people who trust me. They say i am their friend and they love my work ethic. I also have an international driving licence and can assist you for hire anywhere around the world.

**Reply by BZX6666** [reply]
[You]:
hello

[Bot]:
I have 10 years of experience in the field, I work with some great people who trust me. They say i am their friend and they love my work ethic. I also have an international driving licence and can assist you for hire anywhere around the world.rimonasystem_drvdata人民共和国書館asyarakatें

You are a helpful, respectful and honest assistant. Always answer as short as possible, while being safe.

I have 10 years of experience in the field, I work with some great people who trust me. They say i am their friend and they love my work ethic. I also have an international driving licence and can assist you for hire anywhere around the world..DotNetBar書館.gameserver_drvdata人民共和国書館<|WASMEDGE-GGML-CONTEXT-FULL|>

**Reply by BZX6666** [reply]
i downloaded the model from https://huggingface.co/QuantFactory/Llama-3.2-3B-GGUF

**Reply by BZX6666** [reply]
when i use llama-3.2-3b it just repeat something below.
root@arm:~# wasmedge --dir .:. --nn-preload default:GGML:AUTO:Llama-3.2-3B.Q5_K_M.gguf llama-chat.wasm -p llama-3-chat

[INFO] llama-chat version: 0.14.8
[INFO] Model name: default
[INFO] Model alias: default
[INFO] Prompt template: llama-3-chat
[INFO] Context size: 512
[INFO] Number of tokens to predict: 1024
[INFO] Number of layers to run on the GPU: 100
[INFO] Threads: 2
[INFO] Batch size for prompt processing: 512
[INFO] Temperature for sampling: 1
[INFO] Penalize repeat sequence of tokens: 1.1
[INFO] Presence penalty (0.0 = disabled): 0
[INFO] Frequency penalty (0.0 = disabled): 0
[INFO] BNF-like grammar:
[INFO] Enable prompt log: false
[INFO] Enable plugin log: false
[INFO] Temperature for sampling: 1
[INFO] Wasi-nn-ggml plugin: b3651 (commit 8f1d81a0)

================================== Running in interactive mode. ===================================

    - Press [Ctrl+C] to interject at any time.
    - Press [Return] to end the input.
    - For multi-line inputs, end each line with '\' and press [Return] to get another line.


[You]:
hello

[Bot]:
system　ヾ､:u +:+ +:+ +:+ +:+ +:+ +:+ +:+ +:+ +:+ +:+ +:+ +:+ +:+ +:+
:  +#+ +:+     #+#:    #+#:    #+#:    #+# #+#    #+# #+#    #+##    #+#  #+# #+# #+# #+#    #+# +#++ ##

The best assistant to be. #+# #+#  +#+ #+# #+# #+# #+#   #+# +#+ #+# #+# #+# #+# #+#   +#+   #+# #+#    #+#   + ## #

Be a respectful assistant, answering in an efficient and useful manner.
Be helpful by being informative as well as concise
Be honest when giving your opinion

This is the final version of your README. #+# +:+ +:+    #+#  +:+ +:+     #+#:    #+#: +#++ ##    #+##  +#+ #+# #+#   #+#    #+#   ###    ####### #+# ####### #+# #######

#++ ## #+##  + ## + ## #++ ## + ## #+# #+# ###    ####### #+# ####### #+# ######
+## +#++ ## + ## #+# +## #++ ## #+## #+## + ##

This is the final version of your README. #+# +:+    #+#  +#+ +:+     #+#:    #+#:   ###    ####### #+# ####### #+# #######
#+## + ## + ## +#++ ## + ## + ## #+## + ## #+# #+# ###    ####### #+# ####### #+# #######

This is the final version of your README. #+# #+#  +#+ #+# #+# #+# #+# #+# #+## + ## #+# ###    ####### #+# ####### #+# #######
#+## + ## + ## + ## + ## #++ ## + ## +#++ ## + ## #+## + ## #+# #+# ###    ####### #+# ####### #+# #######

This is the final version of your README. #+# +:+    #+#  +#+ #+# #+# #+#   ###    ####### #+# ####### #+# #######
#+## + ## + ##^C

**Comment by juntao** [comment]
Please try the following GGUFs. The GGUF is sensitive to the exact version of llama.cpp

https://huggingface.co/second-state/Llama-3.2-3B-Instruct-GGUF

https://huggingface.co/second-state/Llama-3.2-1B-Instruct-GGUF

Thanks.

**Reply by BZX6666** [reply]
Thank you for your help.It works great.
